### PR TITLE
In Cassandra3 storage, only use consistency level LOCAL_ONE if a local_dc has been defined, otherwise use ONE

### PR DESCRIPTION
**Pull request by michaelsembwever**
Also…

 - Mark all writes as idempotent.
 - Update the retry policy, honouring idempotence, consistency level, and applying appropriate back-pressure.

### CODE REVIEW

File: zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java [hunk 1]
```diff
@@ -123,6 +125,13 @@ static Cluster buildCluster(Cassandra3Storage cassandra) {
     builder.withPoolingOptions(new PoolingOptions().setMaxConnectionsPerHost(
         HostDistance.LOCAL, cassandra.maxConnections
     ));
+
+    builder.withQueryOptions(
```
Comment (codefromthecrypt) [inline zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java:129]: this is the only one I don't know just by reading it. Can you put a comment line of why we do this, or tell me and I'll add it? Thx!

**Review by codefromthecrypt — APPROVED** [review]
reformatted w/ https://github.com/square/java-code-styles vs nagging you :)

just one thing on my end, just to understand this. @llinder how do you feel?

**Review by llinder — APPROVED** [review]

File: zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java (hunk 1 above)
Comment (michaelsembwever) [reply zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java:129]: if `local_dc` isn't defined then LOCAL_ONE sticks requests to the dc of the first seed host that the driver connects to.
ONE will at least in this situation connect to any dc, and then any latency awareness will bring it back to (eventually) using the local dc.

furthermore `LOCAL_ONE` can fail easily if you're not using 'NetworkTopologyStrategy` but that's a fix for another day.

Comment (michaelsembwever) [reply zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java:129]: added comment. (can you please squash the 4 commits!)

Comment (codefromthecrypt) [reply zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/DefaultSessionFactory.java:129]: great info. ty

**Comment by codefromthecrypt** [comment]
thanks tons!

**Pull request merged**